paasta_tools/drain_lib.py

```diff
@@ -262,25 +263,56 @@ def spool_url(self, task: DrainTask) -> str:
                     info['reason'] = groupdict['reason']
                 return info
 
+    async def for_each_registration(self, task: DrainTask, func: Callable[..., T]) -> asyncio.Future[T]:
+        futures = [
+            func(url)
+            for url in self.spool_urls(task)
+        ]
+        return await asyncio.gather(*futures)
+
     async def drain(self, task: DrainTask) -> None:
-        return await self.post_spool(task, 'down')
+        await self.for_each_registration(
```

Comment:
With these functions, they used to "return" the await function thingy, but now they will return None, is that OK?

Reply:
Yeah, it was just returning None anyway (see the type signature)